### PR TITLE
[E2E] Tests for Docker

### LINKED ISSUE #3355: [E2E] port integration test "docker" to the e2e framework
## E2E Task
Port:
- https://github.com/sylabs/singularity/blob/master/cmd/singularity/docker_test.go
to the e2e framework:
 - https://github.com/sylabs/singularity/tree/master/e2e
under `e2e/docker`

### PR DESCRIPTION
**Pull request by WestleyK**
## Description of the Pull Request (PR):

This PR adds e2e tests for docker_test.go.

## This fixes or addresses the following GitHub issues:

- Fixes #3355 

<br>

### CODE REVIEW

File: e2e/suite.go [hunk 1]
```diff
@@ -90,6 +91,7 @@ func Run(t *testing.T) {
 	defer os.Remove(imagePath)
 
 	// RunE2ETests by functionality
-	t.Run("BUILD", imgbuild.RunE2ETests)
-	t.Run("ACTIONS", actions.RunE2ETests)
+//	t.Run("BUILD", imgbuild.RunE2ETests)
+//	t.Run("ACTIONS", actions.RunE2ETests)
```
Comment (ArangoGutierrez) [inline e2e/suite.go]: un-comment

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

**Comment by ArangoGutierrez** [comment]
could you add a `deprecated` note on https://github.com/sylabs/singularity/blob/master/cmd/singularity/docker_test.go
```
// This file is been deprecated and will disappear on with version 3.3
// of singularity. The functionality has been moved to e2e/docker/docker.go
```

**Comment by ArangoGutierrez** [comment]
and make golanglint happy :) first

File: e2e/suite.go (hunk 1 above)
Comment (WestleyK) [reply e2e/suite.go]: Yeah, I comment that so I can test faster, still a WIP... :+1:

**Comment by ArangoGutierrez** [comment]
PING me when ready @WestleyK

**Comment by ArangoGutierrez** [comment]
ready @WestleyK ?

File: e2e/imgbuild/imgbuild_utils.go [hunk 2]
```diff
@@ -25,7 +25,8 @@ type Opts struct {
 	Env     []string
 }
 
-type defFileDetail struct {
+// DefFileDetail ...
```
Comment (ArangoGutierrez) [inline e2e/imgbuild/imgbuild_utils.go]: hahahahaha

**Review by ArangoGutierrez — APPROVED** [review]

**Pull request merged**